Adding 0.8.0 release to the docs (#2052)
NVIDIA/cuda-quantum: docs/sphinx/releases.rst

- `Documentation <https://nvidia.github.io/cuda-quantum/latest>`__
- `Examples <https://github.com/NVIDIA/cuda-quantum/tree/main/docs/sphinx/examples>`__

**0.8.0**

The 0.8.0 release adds a range of changes to improve the ease of use and performance with CUDA-Q. 

> The changes listed below highlight some of what we think will be most useful features and changes 

Reviewer: ```suggestion
The changes listed below highlight some of what we think will be the most useful features and changes 
```

> to know about. While the changes list do not capture all of the great contributions, we would like 

Reviewer: ```suggestion
to know about. While the changes list does not capture all of the great contributions, we would like 
```

> to extend many thanks for every contribution in particular also to external contributors.

Reviewer: ```suggestion
to extend many thanks for every contribution, in particular those from external contributors.
```
